source/agora/network/Manager.d

```diff
@@ -1063,6 +1066,44 @@ public class NetworkManager
         return headers_updated;
     }
 
+    /// Retrieve all the missing enrollments from the connected nodes
+    public void getEnrollments (EnrollmentManager enroll_man,
+        scope UTXOFinder finder) @safe nothrow
+    {
+        NodeConnInfo[] conns = this.peers.array;
+        ulong[] conn_indices = conns.length.iota.array;
```

Comment:
So this essentially creates an array being `[0, 1, 2, 3, ...]` ?
And it *only* uses it for a `foreach`. We should try to avoid pointless memory allocations like this.
A simple way to do this is:
```D
foreach (idx; 0 .. conns.length)
```

But then, the only use of `idx` in that loop is for `conns[idx]`, so why not simply:
```D
foreach (conn; this.peers)
```

And avoid two superfluous memory allocations ?

Reply:
Updated.